module/zfs/dnode.c

```diff
@@ -436,11 +436,15 @@ dnode_verify(dnode_t *dn)
 	if (dn->dn_phys->dn_type != DMU_OT_NONE)
 		ASSERT3U(dn->dn_phys->dn_nlevels, <=, dn->dn_nlevels);
 	ASSERT(DMU_OBJECT_IS_SPECIAL(dn->dn_object) || dn->dn_dbuf != NULL);
+#ifdef DEBUG
 	if (dn->dn_dbuf != NULL) {
+		mutex_enter(&dn->dn_dbuf->db_mtx);
```

Comment:
meta-dnode dbufs are not relocatable, and we only compare pointers. It does not need locking.